controllers/topics/broadcastMessage.js

```diff
@@ -0,0 +1,117 @@
+/* 
+    Send message to all users who join a topic
+*/
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const UsersFunction = require('../../databases/functions/users');
+const {followTopicQueue} = require('../../services/redis/queue');
+const {User, CommunityMessageFormat, UserTopic} = require('../../databases/models');
+const {Op} = require('sequelize');
+
+const sendMessageToQueue = async (user_id, topic_id, community_message_format_id, delay) => {
+  const data = {
+    user_id,
+    topic_id,
+    community_message_format_id
+  };
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz().tz('America/Los_Angeles').add(delay, 'minutes');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling in this section is not robust. Simply logging the error to the console may not be sufficient for production-level code. Consider throwing the error or handling it in a way that's more visible and actionable.
```diff
-    console.log('error', e);
+    throw new Error(e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->